lib/array_equals.rb

```diff
@@ -1,5 +1,23 @@
-# Determines if the two input arrays have the same count of elements
-# and the same integer values in the same exact order
+
 def array_equals(array1, array2)
-  raise NotImplementedError
+  i = 0 
+  boolean_value = true 
+
+  if array1 == nil && array2 == nil
```

Comment:
this could be simiplified a bit with:

```ruby
if array1.nil? && array2.nil?
  return true
elsif array1.nil? || array2.nil? # if one or the other is nil, but not both
  return false
```